doc/credentials.md

```diff
@@ -0,0 +1,230 @@
+# Credentials
+Camera credentials are stored in the EdgeX Secret Store and referenced by MAC Address. All 
+devices by default are configured with credentials from `DefaultSecretPath` unless configured
+as part of a group within `AppCustom.CredentialsMap`.
+
+Three things must be done in order to add an authenticated camera to EdgeX:
+- Add device to EdgeX
+  - Manually or via auto-discovery
+- Add `Credentials` to `Secret Store`
+  - Manually or via utility scripts
+- Map `Credentials` to devices
+  - Manually or via utility scripts
+  - Configure as `DefaultSecretPath`
+
+## Terminology / Definitions
+- **Secret**: A generic object which is stored as key/value pairs in the `Secret Store` under a specific `Secret Path` key.
+- **Credentials**: A specific type of `Secret` which contains a mapping of `username`, `password`, and authentication `mode`.
+- **Secret Store**: The place EdgeX stores all `Secrets`
+  - In secure mode this is `Vault`
+  - In non-secure mode this is the configuration provider (typically `Consul`). 
+    They can be pre-configured via `configuration.toml`'s  `Writable.InsecureSecrets` section.
+- **Secret Path**: The name/key of the `Secret` as they are stored in the `Secret Store`.
+- **CredentialsMap**: (aka `AppCustom.CredentialsMap`) this contains the mappings between `Secret Path` and
+    `MAC Address`. Each key in the map is a `Secret Path` which points to `Credentials` in the `Secret Store`. The value
+    for each key is a comma separated list of `MAC Address`es which should use those `Credentials`.
```

Comment:
` needs to be updated

Reply:
yeah i had that on purpose because the actual thing is the `Mac Address`, but it does look funky leaving the "es" out of it. I will fix.

Reply:
@ajcasagrande Even I preferred that way it is now as it implies one or more mac addresses